sdk/src/main/java/io/horizen/account/state/MessageProcessor.java

```diff
@@ -14,10 +14,10 @@
 public interface MessageProcessor {
     // Initialization is going to happen only once at genesis State creation.
     // Common pattern: declare a new native smart contract account in the View
-    void init(BaseAccountStateView view) throws MessageProcessorInitializationException;
+    void init(AccountStateView view) throws MessageProcessorInitializationException;
```

Comment:
MessageProcessors must not access to MetadataStorage. The problem is that MetadataStorage doesn't have snapshots, so in case a transaction fails when a block is applied, there is no way to roll back to the previous snapshot.

Reply:
You are right, I am now using the consensus epoch number in the message processor interfaces. Consider that `process() ` method already has this info in the block context obj parameter